pkg/api/unversioned/time_proto.go

```diff
@@ -39,20 +33,18 @@ type Timestamp struct {
 	// Non-negative fractions of a second at nanosecond resolution. Negative
 	// second values with fractions must still have non-negative nanos values
 	// that count forward in time. Must be from 0 to 999,999,999
-	// inclusive.
+	// inclusive. This field may be limited in precision depending on context.
 	Nanos int32 `json:"nanos"`
 }
 
-// ProtoTime returns the Time as a new ProtoTime value.
-func (m *Time) ProtoTime() *ProtoTime {
+// Timestamp returns the Time as a new Timestamp value.
+func (m *Time) ProtoTime() *Timestamp {
 	if m == nil {
-		return &ProtoTime{}
+		return &Timestamp{}
 	}
-	return &ProtoTime{
-		Timestamp: Timestamp{
-			Seconds: m.Time.Unix(),
-			Nanos:   int32(m.Time.Nanosecond()),
-		},
+	return &Timestamp{
+		Seconds: m.Time.Unix(),
```

Comment:
Not sure - https://golang.org/pkg/time/#Time.Unix

On Tue, Apr 5, 2016 at 6:40 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/api/unversioned/time_proto.go
> <https://github.com/kubernetes/kubernetes/pull/23871#discussion_r58625800>
> :
>
> >  	}
> > -	return &ProtoTime{
> > -		Timestamp: Timestamp{
> > -			Seconds: m.Time.Unix(),
> > -			Nanos:   int32(m.Time.Nanosecond()),
> > -		},
> > +	return &Timestamp{
> > +		Seconds: m.Time.Unix(),
>
> Does this account for leap seconds?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23871/files/1f012a2bb3e07a53b5b28bb9bc8781667508faeb#r58625800>
>